pkg/provisioner/ironic/ironic.go

```diff
@@ -432,16 +432,23 @@ func (p *ironicProvisioner) changeNodeProvisionState(ironicNode *nodes.Node, opt
 	changeResult := nodes.ChangeProvisionState(p.client, ironicNode.UUID, opts)
 	switch changeResult.Err.(type) {
 	case nil:
+		success = true
 	case gophercloud.ErrDefault409:
```

Comment:
If the reason for failure (or at least "not success") is that we need to retry, why wouldn't we need to check that on every call?